presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -111,19 +117,30 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
         NodeMap nodeMap = this.nodeMap.get().get();
         NodeAssignmentStats assignmentStats = new NodeAssignmentStats(nodeTaskMap, nodeMap, existingTasks);
 
-        ResettableRandomizedIterator<InternalNode> randomCandidates = getRandomCandidates(maxTasksPerStage, nodeMap, existingTasks);
+        NodeSelection randomNodeSelection = new RandomNodeSelection(nodeMap, includeCoordinator, minCandidates, maxTasksPerStage, existingTasks);
         Set<InternalNode> blockedExactNodes = new HashSet<>();
         boolean splitWaitingForAnyNode = false;
-        for (Split split : splits) {
-            randomCandidates.reset();
+        List<HostAddress> sortedCandidates = NodeSelector.sortedNodes(nodeMap);
```

Comment:
Sort only if `if (splits.stream().anyMatch(split -> split.getNodeSelectionStrategy() != NO_PREFERENCE))`

Reply:
Shixuan and I had a discussion offline, the splits number may be way bigger than nodes number, so iterating through splits is not necessarily faster than sorting.
Will leave a todo and will think of a way to fix it once we can confirm a bottleneck this causes